cli/rt/41_prompt.js

```diff
@@ -0,0 +1,80 @@
+// Copyright 2018-2020 the Deno authors. All rights reserved. MIT license.
+((window) => {
+  const { stdin, stdout } = window.__bootstrap.files;
+  const { isatty } = window.__bootstrap.tty;
+  const LF = "\n".charCodeAt(0);
+  const encoder = new TextEncoder();
+  const decoder = new TextDecoder();
+
+  function alert(message) {
+    if (!isatty(stdin.rid)) {
+      throw new Error(
+        "stdin is not interactive. 'alert' needs stdin to be interactive.",
+      );
+    }
+
+    if (message) {
+      stdout.writeSync(encoder.encode(message));
+    }
+
+    readLineFromStdinSync();
+  }
+
+  function confirm(message) {
+    if (!isatty(stdin.rid)) {
+      throw new Error(
+        "stdin is not interactive. 'confirm' needs stdin to be interactive.",
+      );
+    }
+
+    if (message) {
+      stdout.writeSync(encoder.encode(message));
+    }
+
+    stdout.writeSync(encoder.encode(" [y/N] "));
+
+    const answer = readLineFromStdinSync();
+
+    return answer === "Y" || answer === "y";
+  }
+
+  function prompt(message, defaultValue = "") {
+    if (!isatty(stdin.rid)) {
+      throw new Error(
+        "stdin is not interactive. 'prompt' needs stdin to be interactive.",
+      );
+    }
```

Comment:
```suggestion
    if (!isatty(stdin.rid)) {
      return null;
    }
```

Reply:
How about returning `defaultValue` here?

Reply:
I got firefox to disable dialog boxes and tried this with a `defaultValue`, it returned `null`. I guess `defaultValue` is strictly the user interface default.